File: sdk/java/picodroid/os/SystemClock.java
// SPDX-License-Identifier: GPL-3.0-only
package picodroid.os;

public class SystemClock {
  public static native void sleep(int ms);

  public static native long elapsedRealtimeNanos();

  /**
   * Anchors the wall clock: after this call {@code System.currentTimeMillis()} returns real epoch
   * time (before any call it counts from boot). Typically fed from an SNTP sync. Always returns
   * {@code true} — Android's permission-denied case does not apply here.
   */
  public static native boolean setCurrentTimeMillis(long millis);
}

File: picodroid-esp/src/hal/esp/system_clock.rs
// SPDX-License-Identifier: GPL-3.0-only
// ESP32-S3 system clock stubs — Milestone 1.
// Real delay via esp-hal and elapsed time via RTC are Milestone 2.

pub fn sleep(_ms: u32) {}

pub fn elapsed_realtime_nanos() -> i64 {
    0
}
